### PR TITLE
 refactor: update `blas/ext/base/ssorthp` to follow current project conventions

### LINKED ISSUE #1537: [RFC]: refactor `blas/ext/base/ssorthp` to follow current project conventions
### Description

This RFC proposes refactoring [`@stdlib/blas/ext/base/ssorthp`](https://github.com/stdlib-js/stdlib/tree/develop/lib/node_modules/%40stdlib/blas/ext/base/ssorthp) to follow current project conventions.

These conventions are outlined in more detail in https://github.com/stdlib-js/stdlib/issues/788. Namely, we want to migrate from C++ add-on interfaces to C add-on interfaces and then make various style and simplification changes.

Some example commits which perform most, if not all, of the requested changes include https://github.com/stdlib-js/stdlib/commit/d66d80cfcbab1da95cb5ae8cfae4a1fc4c5129d1 (saxpy),  https://github.com/stdlib-js/stdlib/commit/3c82fe7b6c0f9fd6ef02c86d6e2024d708ac8c14 (cswap), and https://github.com/stdlib-js/stdlib/commit/5b15c9675be434c0e11b1ec1e1f78d7b089abbcd (dasum).

### Related Issues

Related issues #788, #1152.

### Questions

No.

### Other

No.

### Checklist

- [X] I have read and understood the [Code of Conduct](https://github.com/stdlib-js/stdlib/blob/develop/CODE_OF_CONDUCT.md).
- [X] Searched for existing issues and pull requests.
- [X] The issue name begins with `RFC:`.

### PR DESCRIPTION
**Pull request by naveen1m**
Resolves #1537

## Description

> What is the purpose of this pull request?

This pull request:

-   Refactors `@stdlib/blas/ext/base/ssorthp` to follow current project conventions.

## Checklist
- [x] update README.md
- [x] manifest.json
- [x] include.gypi
- [x] benchmark 
- [x] examples
- [x] docs
- [x] test
- [x] src
- [x] lib

## Related Issues

> Does this pull request have any related issues?

This pull request:

-   resolves #1537 
-   progress #1152 

## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

No.

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

-   [x] Read, understood, and followed the [contributing guidelines][contributing].

* * *

@stdlib-js/reviewers

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

### CODE REVIEW

**Review by Pranavchiku — CHANGES_REQUESTED** [review]
This PR needs more changes, please follow https://github.com/stdlib-js/stdlib/commit/d66d80cfcbab1da95cb5ae8cfae4a1fc4c5129d1 and add required files.

File: lib/node_modules/@stdlib/blas/ext/base/ssorthp/README.md [hunk 1]
```diff
@@ -50,40 +50,38 @@ The function has the following parameters:
...
-var N = floor( x.length / 2 );
 
-ssorthp( N, -1.0, x, 2 );
+ssorthp( 2, -1.0, x, 2 );
 // x => <Float32Array>[ 3.0, -2.0, 1.0, -4.0 ]
 ```
 
 Note that indexing is relative to the first index. To introduce an offset, use [`typed array`][mdn-typed-array] views.
 
+
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/blas/ext/base/ssorthp/README.md]:
```suggestion
```

File: lib/node_modules/@stdlib/blas/ext/base/ssorthp/README.md [hunk 2]
```diff
@@ -135,27 +133,13 @@ ssorthp.ndarray( 3, 1.0, x, 1, x.length-3 );
...
-    sign = randu();
-    if ( sign < 0.5 ) {
-        sign = -1.0;
-    } else {
-        sign = 1.0;
-    }
-    x[ i ] = sign * rand;
-}
+var rand = discreteUniform( -100, 100 );
+var x = filledarrayBy( 10, 'float32', rand);
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/blas/ext/base/ssorthp/README.md]:
```suggestion
var x = filledarrayBy( 10, 'float32', rand );
```

File: lib/node_modules/@stdlib/blas/ext/base/ssorthp/include.gypi [hunk 3]
```diff
@@ -1,6 +1,6 @@
 # @license Apache-2.0
 #
-# Copyright (c) 2020 The Stdlib Authors.
+# Copyright (c) 2024 The Stdlib Authors.
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/blas/ext/base/ssorthp/include.gypi]:
```suggestion
# Copyright (c) 2020 The Stdlib Authors.
```

**Review by Planeshifter — CHANGES_REQUESTED** [review]
Please have a look at the failing CI job for linting of the C files.

Also, the PR currently took the instruction to "remove the use of floor everywhere" as replacing it via a a right shift operation. This will have to be changed; please consult the linked packages for the changes we are looking for.

Thanks for your efforts!

**Comment by naveen1m** [comment]
Got it! thanks for the clarification. I will make the necessary changes.

File: lib/node_modules/@stdlib/blas/ext/base/ssorthp/benchmark/benchmark.mostly_sorted_few_uniques.js [hunk 4]
```diff
@@ -60,7 +59,7 @@ function createBenchmark( iter, len ) {
 		for ( j = 0; j < len; j++ ) {
 			sa = (b-a) * (j/len);
 			sb = sa / 2.0;
-			tmp[ j ] = floor( uniform( a+sa, b+sb ) );
+			tmp[ j ] = 2*( uniform( a+sa, b+sb ) )>>1;
```
Comment (kgryte) [inline lib/node_modules/@stdlib/blas/ext/base/ssorthp/benchmark/benchmark.mostly_sorted_few_uniques.js]: This isn't where you should be removing "floor". You are only supposed to remove from documentation and tests. E.g.,

```
N = floor( x.length / 2 );
```

Those are the only instances. Not here in the benchmarks.

**Comment by naveen1m** [comment]
@kgryte @Planeshifter can you please review it? 
 
this test is failing, while other file has included same directory [see](https://github.com/stdlib-js/stdlib/commit/d66d80cfcbab1da95cb5ae8cfae4a1fc4c5129d1#diff-af83a9a228abf78533a2b199ce69ed7b06bece7905eebc62ca899fe9ab1c032f)

Screenshot 2024-04-09 at 6 13 13 AM
run_affected_tests

**Comment by Planeshifter** [comment]
Tests should now pass; will merge once CI has passed. Thanks!

**Review by Planeshifter — APPROVED** [review]

**Pull request merged**